extensions/registries/src/extension.ts

```diff
@@ -0,0 +1,74 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as extensionApi from '@tmpwip/extension-api';
+import * as fs from 'node:fs';
+
+// The image path for the registry logos
+const imagePath = __dirname + '/images/';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+  // For each defaultRegistries, suggest the registry to Podman Desktop
+  defaultRegistries.forEach(registry => {
+    extensionApi.registry.suggestRegistry(registry);
+  });
+}
+
+export function deactivate(): void {
+  // For each defaultRegistries, suggest the registry to Podman Desktop
+  defaultRegistries.forEach(registry => {
+    extensionApi.registry.unsuggestRegistry(registry);
+  });
+
+  console.log('stopping registries extension and clearing suggested registries');
+}
+
+// Const array of list of approved registries that contain the default URL as well as base64 encoded version of their logo
+const defaultRegistries: extensionApi.RegistrySuggestedProvider[] = [
+  {
+    name: 'Docker Hub',
+    url: 'docker.io',
+    icon: base64_encode(imagePath + 'docker.png'),
+  },
+  {
+    name: 'Red Hat Quay',
+    url: 'quay.io',
+    icon: base64_encode(imagePath + 'quay.png'),
+  },
+  {
+    name: 'GitHub',
+    url: 'ghcr.io',
+    icon: base64_encode(imagePath + 'github.png'),
+  },
+  {
+    name: 'IBM Container Registry',
+    url: 'icr.io',
+    icon: base64_encode(imagePath + 'ibm.png'),
+  },
+  {
+    name: 'Google Container Registry',
+    url: 'gcr.io',
+    icon: base64_encode(imagePath + 'gcr.png'),
+  },
```

Comment:
here I don't know as gcr is replaced by artifact registry and if there is Google registry, probably we should have amazon and azure one as well

Reply:
yeah, maybe remove IBM suggestion, add amazon and azure?

Reply:
I think that we may need some enhancement on the suggesting API as for some, we need extra information

for example with Azure URLs are like `MY-CUSTOM-NAME.azurecr.io` (URL azurecr.io is not used directly)

so maybe we start with very few like GitHub, DockerHub and Quay.io and wait feedback
If people have other thoughts as well here.

Reply:
Same with GitLab, so can't add it just yet (the docs show: registry.gitlab.example.com) https://docs.gitlab.com/ee/user/packages/container_registry/

You're right. Let's just do Quay, Google, Github and Docker for now.